# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/x11/xfree86/4.3.11-config.patch
# ROCK Linux is Copyright (C) 1998 - 2003 Clifford Wolf
# 
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version. A copy of the GNU General Public
# License can be found at Documentation/COPYING.
# 
# Many people helped and are helping developing ROCK Linux. Please
# have a look at http://www.rocklinux.org/ and the Documentation/TEAM
# file for details.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---


RCS file: /cvs/xc/programs/Xserver/hw/xfree86/common/xf86Config.c,v
retrieving revision 3.272
retrieving revision 3.273
diff -u -r3.272 -r3.273
--- xc/programs/Xserver/hw/xfree86/common/xf86Config.c	24 Aug 2003 20:52:30 -0000	3.272
+++ xc/programs/Xserver/hw/xfree86/common/xf86Config.c	27 Aug 2003 02:50:25 -0000	3.273
@@ -1,4 +1,4 @@
-/* $XFree86: xc/programs/Xserver/hw/xfree86/common/xf86Config.c,v 3.272 2003/08/24 20:52:30 dawes Exp $ */
+/* $XFree86: xc/programs/Xserver/hw/xfree86/common/xf86Config.c,v 3.273 2003/08/27 02:50:25 dawes Exp $ */
 
 
 /*
@@ -1451,7 +1451,7 @@
 	indp[count - 1].extraOptions = xf86addNewOption(NULL, "CorePointer", NULL);
 	indp[count].identifier = NULL;
 	servlayoutp->inputs = indp;
-    } else {
+    } else if (!havePointer) {
 	/* This should never happen. */
 	xf86Msg(X_ERROR, "Cannot locate a core pointer device.\n");
 	return FALSE;
@@ -1473,7 +1473,7 @@
 	indp[count - 1].extraOptions = xf86addNewOption(NULL, "CoreKeyboard", NULL);
 	indp[count].identifier = NULL;
 	servlayoutp->inputs = indp;
-    } else {
+    } else if (!haveKeyboard) {
 	/* This should never happen. */
 	xf86Msg(X_ERROR, "Cannot locate a core keyboard device\n");
 	return FALSE;


